network-store-client/src/main/java/com/powsybl/network/store/client/BufferedRestNetworkStoreClient.java

```diff
@@ -77,23 +101,64 @@ public void createNetworks(List<Resource<NetworkAttributes>> networkResources) {
         return client.getNetwork(networkUuid);
     }
 
+    private <T extends IdentifiableAttributes> List<Resource<T>> adapt(List<Resource<T>> listResources) {
+        listResources.forEach(resource -> resource.setStoreClient(this));
+        return listResources;
+    }
+
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private <T extends IdentifiableAttributes> Optional<Resource<T>> adapt(Optional<Resource<T>> resource) {
+        if (resource.isPresent()) {
+            resource.get().setStoreClient(this);
+        }
+        return resource;
+    }
+
+    private static <T extends IdentifiableAttributes> void update(List<Resource<T>> updatelistResources, List<Resource<T>> listResources) {
+        List<Resource<T>> resourcesToRemove = new ArrayList<>();
+        for (Resource<T> resource1 : updatelistResources) {
```

Comment:
Is it ok to have O(n²) algo here ? can't we have 10000 x 10000 elements ? Otherwise make sets or sort by ids

Reply:
In practice, the second argument list will always contain one element, so there is no really performance problem here, I think.
But, effectively, we could change the type of the different update...ResourcesToFlush fields to use a set or a map by resource id for the values instead of a simple list ...

Reply:
The code has been refactored, so the second argument is now a Resource<T> object